specs/latest/2.0/index.html

```diff
@@ -768,22 +827,51 @@ <h2 class="no-toc">Table of contents</h2>
+  void uniform4iv(WebGLUniformLocation? location, Int32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+
+  void uniform1fv(WebGLUniformLocation? location, Float32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+  void uniform2fv(WebGLUniformLocation? location, Float32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+  void uniform3fv(WebGLUniformLocation? location, Float32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+  void uniform4fv(WebGLUniformLocation? location, Float32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+
+  void uniform1uiv(WebGLUniformLocation? location, Uint32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+  void uniform2uiv(WebGLUniformLocation? location, Uint32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+  void uniform3uiv(WebGLUniformLocation? location, Uint32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+  void uniform4uiv(WebGLUniformLocation? location, Uint32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+
+  void uniformMatrix2fv(WebGLUniformLocation? location, GLboolean transpose, Float32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+  void uniformMatrix3x2fv(WebGLUniformLocation? location, GLboolean transpose, Float32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+  void uniformMatrix4x2fv(WebGLUniformLocation? location, GLboolean transpose, Float32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+
+  void uniformMatrix2x3fv(WebGLUniformLocation? location, GLboolean transpose, Float32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+  void uniformMatrix3fv(WebGLUniformLocation? location, GLboolean transpose, Float32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+  void uniformMatrix4x3fv(WebGLUniformLocation? location, GLboolean transpose, Float32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+
+  void uniformMatrix2x4fv(WebGLUniformLocation? location, GLboolean transpose, Float32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+  void uniformMatrix3x4fv(WebGLUniformLocation? location, GLboolean transpose, Float32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+  void uniformMatrix4fv(WebGLUniformLocation? location, GLboolean transpose, Float32Array data, optional unsigned long srcOffset = 0, optional unsigned long srcLength = 0);
+
```

Comment:
What is the rationale for having a srcLength parameter on these uniformXXX functions?  The WebGL implementation, obviously, internally knows the size of the uniform. When would someone store portions of a single uniform in non-contiguous ArrayBuffers?  Especially for uniform1Xv, this doesn't make sense to me.